network_simulator/plot_tool.py

```diff
@@ -0,0 +1,300 @@
+        plt.legend(sent_legend)
+        # Set up labels.
+        plt.xlabel("Time (sec)")
+        plt.ylabel("Packet Sent Rate")
+        # Set up grid on graph.
+        plt.grid(True)
+
+        # Finalize receive graph.
+        plt.subplot(435)
+        plt.legend(receive_legend)
+        # Set up labels.
+        plt.xlabel("Time (sec)")
+        plt.ylabel("Packet Received Rate")
+        # Set up grid on graph.
+        plt.grid(True)
+
+        # Finalize RTT graph.
+        plt.subplot(438)
+        plt.legend(rtt_legend)
+        # Set up labels.
+        plt.xlabel("Time (sec)")
+        plt.ylabel("RTT (sec)")
+        # Set up grid on graph.
+        plt.grid(True)
+
+        # Finalize window size graph.
+        plt.subplot(4,3,11)
+        plt.legend(window_legend)
+        # Set up labels.
+        plt.xlabel("Time (sec)")
+        plt.ylabel("Window Size (pkts)")
+        # Set up grid on graph.
+        plt.grid(True)
+
+    def graph_hosts(self):
+        """
+        Graphs the host stats of each available host.
+        """
+        # Set up legend storage for host names.
+        sent_legend = []
+        receive_legend = []
+
+        # Iterate through each host to display each stats.
+        for host, host_stats in self.stats.host_stats.iteritems():
+            # Name of host would be "host_address".
+            # Packet sent times.
+            plt.subplot(433)
+            sent_rate_lst = self.tool.gen_rate_tuple_list(
+                host_stats.packet_sent_times)
+            self.tool.graph_tuple_list(sent_rate_lst)
+            sent_legend.append(host)
+
+            # Packet receive times.
+            plt.subplot(436)
+            rec_rate_lst = self.tool.gen_rate_tuple_list(
+                host_stats.packet_rec_times)
+            self.tool.graph_tuple_list(rec_rate_lst)
+            receive_legend.append(host)
+
+        # Finalize sent graph.
+        plt.subplot(433)
+        plt.legend(sent_legend)
+        # Set up labels.
+        plt.xlabel("Time (sec)")
+        plt.ylabel("Packet Sent Rate")
+        # Set up grid on graph.
+        plt.grid(True)
+
+        # Finalize receive graph.
+        plt.subplot(436)
+        plt.legend(receive_legend)
+        # Set up labels.
+        plt.xlabel("Time (sec)")
+        plt.ylabel("Packet Received Rate")
+        # Set up grid on graph.
+        plt.grid(True)
+
+    def graph_network(self):
+        """
+        Graphs the entire network (this includes all links, flows, and hosts.
+        """
+        # Set up matplotlib window name.
+        fig = plt.figure()
+        fig.canvas.set_window_title("Network Statistics")
+
+        # Figure adjust to window size.
+        fig.set_size_inches(35, 10.5, forward=True)
+
+        # Graph all three devices.
+        self.graph_links()
+        self.graph_flows()
+        self.graph_hosts()
+
+        # Make spacing between plots.
+        plt.tight_layout()
+        #plt.axis('equal')
```

Comment:
Could remove this line, or have an inline comment describing what it used to do (inline comments should occur two spaces after the code ends, i.e. "a = foo  # bar").